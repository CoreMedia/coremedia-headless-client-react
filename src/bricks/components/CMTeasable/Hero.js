// @flow
import React from "react";
import PropTypes from "prop-types";

import Picture from "../CMPicture";
import Overlay from "./Overlay";
import OptionalLink from "../Common/OptionalLink";
import Strings from "../../../l18n";
import getColors from "../../../styles/colors";
import {TeaserBox} from "../../../styles";

const composeContent = (url:string, pictureLink: string, pictureTitle: string, pictureAlt: string, title?: string, text?: string, params?: any) => {
  const colors = getColors(params.color);
  const content = [];
  if (pictureLink) {
    content.push(
      <Picture key="picture"
               link={pictureLink}
               ratio="landscape_ratio16x9"
               title={pictureTitle}
               alt={pictureAlt}
               colors={colors}
      />
    );
  }
  if (title || text || params.ctaShow) {
    content.push(
      <Overlay key="overlay"
               fixed={false}
               width={70}
               height="auto"
               bottom="5%"
               title={title}
               text={text}
               colors={colors}
               ctaShow={params.ctaShow && !!url}
               ctaText={params.ctaText || Strings.hero_cta_text}
      />
    );
  }
  return content;
};

type Props = {
  url?: string,
  pictureLink?: string,
  pictureTitle?: string,
  pictureAlt?: string,
  title?: string,
  text?: string,
  params?: any
};

const Hero = ({url, pictureLink, pictureTitle, pictureAlt, title, text, params = {},}: Props) => {
  return (
    <OptionalLink url={url}>
      <TeaserBox>
        {composeContent(url, pictureLink, pictureTitle, pictureAlt, title, text, params)}
      </TeaserBox>
    </OptionalLink>
  );
};

Hero.propTypes = {
  url: PropTypes.string,
  pictureLink: PropTypes.string,
  pictureTitle: PropTypes.string,
  pictureAlt: PropTypes.string,
  title: PropTypes.string,
  text: PropTypes.string,
  params: PropTypes.any
};

export default Hero;
