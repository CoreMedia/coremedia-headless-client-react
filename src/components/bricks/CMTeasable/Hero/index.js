// @flow
import * as React from 'react';
import PropTypes from 'prop-types';

import Picture from '../../CMPicture';
import Overlay from '../Overlay';
import { TeaserBox } from '../../../basic/Box';
import { OptionalLink } from '../../../basic/Link';
import Strings from '../../../../l18n';
import type { Teaser } from '../../../../types';

const composeContent = ({
  url,
  pictureLink,
  pictureTitle,
  pictureAlt,
  title,
  text,
  params,
}: Teaser): React.Node => {
  const color = params && params.color;
  const content = [];
  if (pictureLink) {
    content.push(
      <Picture
        key="picture"
        link={pictureLink}
        ratio="landscape_ratio16x9"
        title={pictureTitle}
        alt={pictureAlt}
        color={color}
      />
    );
  }
  if (title || text || (params && params.ctaShow)) {
    content.push(
      <Overlay
        key="overlay"
        fixed={false}
        width={70}
        height="auto"
        bottom="5%"
        title={title}
        text={text}
        color={color}
        ctaShow={!!params && !!params.ctaShow && !!url}
        ctaText={(params && params.ctaText) || Strings.hero_cta_text}
      />
    );
  }
  return content;
};

const HeroTeaser = ({
  url,
  pictureLink,
  pictureTitle,
  pictureAlt,
  title,
  text,
  params = {},
}: Teaser) => (
  <OptionalLink url={url}>
    <TeaserBox>
      {composeContent({ url, pictureLink, pictureTitle, pictureAlt, title, text, params })}
    </TeaserBox>
  </OptionalLink>
);

HeroTeaser.propTypes = {
  url: PropTypes.string,
  pictureLink: PropTypes.string,
  pictureTitle: PropTypes.string,
  pictureAlt: PropTypes.string,
  title: PropTypes.string,
  text: PropTypes.string,
  params: PropTypes.shape({
    color: PropTypes.string,
    ctaShow: PropTypes.bool,
    ctaText: PropTypes.string,
    url: PropTypes.string,
  }),
};

export default HeroTeaser;
