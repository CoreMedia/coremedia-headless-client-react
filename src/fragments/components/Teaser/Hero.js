import React from "react";

import {Hero as HeroBrick} from "../../../bricks/components/CMTeasable";

const Hero = {
  type: 'teasables',
  view: null,
  brick(data, params) {
    const picture = data.picture || {};
    return (
      <HeroBrick url={params.url ? params.url : data.teaserTarget.link}
                 pictureLink={picture.link}
                 pictureTitle={picture.title}
                 pictureAlt={picture.alt}
                 title={data.teaserTitle}
                 text={data.teaserText}
                 params={params}
      />
    );
  }
};

export default Hero;
