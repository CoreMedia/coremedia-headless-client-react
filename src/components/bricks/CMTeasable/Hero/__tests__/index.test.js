// @flow
import React from 'react';

import Hero from '..';

describe('HeroTeaser Component', () => {
  it('should render HeroTeaser with Link, Picture and Overlay', () => {
    const wrapper = shallow(
      <Hero
        params={{ color: 'blue', ctaShow: true, url: '/caas.html' }}
        pictureAlt="Globe"
        pictureLink="coremedia:///image/2656/data"
        pictureTitle="Globe"
        text="Discover the new CoreMedia Headless Services."
        title="Content Management at a new Scale with CoreMedia CaaS"
        url="/caas.html"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render HeroTeaser with Link, Picture and Overlay', () => {
    const wrapper = shallow(
      <Hero
        params={{ color: 'blue', url: '/caas.html' }}
        pictureAlt="Globe"
        pictureLink="coremedia:///image/2656/data"
        pictureTitle="Globe"
        text="Discover the new CoreMedia Headless Services."
        title="Content Management at a new Scale with CoreMedia CaaS"
        url="/caas.html"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render HeroTeaser with Link, Picture and Overlay', () => {
    const wrapper = shallow(
      <Hero
        params={{ color: 'blue', url: '/caas.html' }}
        pictureAlt="Globe"
        pictureLink="coremedia:///image/2656/data"
        pictureTitle="Globe"
        title="Content Management at a new Scale with CoreMedia CaaS"
        url="/caas.html"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render HeroTeaser with Link, Picture and Overlay', () => {
    const wrapper = shallow(
      <Hero
        params={{ color: 'blue', url: '/caas.html' }}
        pictureAlt="Globe"
        pictureLink="coremedia:///image/2656/data"
        pictureTitle="Globe"
        text="Discover the new CoreMedia Headless Services."
        url="/caas.html"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render HeroTeaser with Link, Picture and Overlay', () => {
    const wrapper = shallow(
      <Hero
        params={{ color: 'blue', ctaShow: true, url: '/caas.html' }}
        pictureAlt="Globe"
        pictureLink="coremedia:///image/2656/data"
        pictureTitle="Globe"
        text="Discover the new CoreMedia Headless Services."
        url="/caas.html"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render HeroTeaser with Link, Picture and Overlay', () => {
    const wrapper = shallow(
      <Hero
        params={{ color: 'blue', ctaShow: true, url: '/caas.html' }}
        pictureAlt="Globe"
        pictureLink="coremedia:///image/2656/data"
        pictureTitle="Globe"
        title="Content Management at a new Scale with CoreMedia CaaS"
        url="/caas.html"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render HeroTeaser with Link, Picture and Overlay', () => {
    const wrapper = shallow(
      <Hero
        params={{ color: 'blue', ctaShow: true, url: '/caas.html' }}
        pictureAlt="Globe"
        pictureLink="coremedia:///image/2656/data"
        pictureTitle="Globe"
        url="/caas.html"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render HeroTeaser with Link and Picture', () => {
    const wrapper = shallow(
      <Hero
        params={{ color: 'blue', url: '/caas.html' }}
        pictureAlt="Globe"
        pictureLink="coremedia:///image/2656/data"
        pictureTitle="Globe"
        url="/caas.html"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render HeroTeaser with Link and Overlay', () => {
    const wrapper = shallow(
      <Hero params={{ color: 'blue', ctaShow: true, url: '/caas.html' }} url="/caas.html" />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render HeroTeaser with Link', () => {
    const wrapper = shallow(
      <Hero params={{ color: 'blue', url: '/caas.html' }} url="/caas.html" />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render HeroTeaser with Overlay', () => {
    const wrapper = shallow(<Hero params={{ color: 'blue', ctaShow: true, url: '/caas.html' }} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render HeroTeaser with Picture', () => {
    const wrapper = shallow(
      <Hero
        params={{ color: 'blue', url: '/caas.html' }}
        pictureAlt="Globe"
        pictureLink="coremedia:///image/2656/data"
        pictureTitle="Globe"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render HeroTeaser with Picture (without params)', () => {
    const wrapper = shallow(
      <Hero pictureAlt="Globe" pictureLink="coremedia:///image/2656/data" pictureTitle="Globe" />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render HeroTeaser with Picture and Overlay', () => {
    const wrapper = shallow(
      <Hero
        params={{ color: 'blue', ctaShow: true, url: '/caas.html' }}
        pictureAlt="Globe"
        pictureLink="coremedia:///image/2656/data"
        pictureTitle="Globe"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
