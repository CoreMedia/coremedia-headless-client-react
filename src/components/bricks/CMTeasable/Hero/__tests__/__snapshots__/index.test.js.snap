// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroTeaser Component should render HeroTeaser with Link 1`] = `
<OptionalLink
  url="/caas.html"
>
  <TeaserBox />
</OptionalLink>
`;

exports[`HeroTeaser Component should render HeroTeaser with Link and Overlay 1`] = `
<OptionalLink
  url="/caas.html"
>
  <TeaserBox>
    <Overlay
      bottom="5%"
      color="blue"
      ctaShow={true}
      ctaText="Learn more"
      fixed={false}
      height="auto"
      key="overlay"
      width={70}
    />
  </TeaserBox>
</OptionalLink>
`;

exports[`HeroTeaser Component should render HeroTeaser with Link and Picture 1`] = `
<OptionalLink
  url="/caas.html"
>
  <TeaserBox>
    <WithTheme(Picture)
      alt="Globe"
      color="blue"
      key="picture"
      link="coremedia:///image/2656/data"
      ratio="landscape_ratio16x9"
      title="Globe"
    />
  </TeaserBox>
</OptionalLink>
`;

exports[`HeroTeaser Component should render HeroTeaser with Link, Picture and Overlay 1`] = `
<OptionalLink
  url="/caas.html"
>
  <TeaserBox>
    <WithTheme(Picture)
      alt="Globe"
      color="blue"
      key="picture"
      link="coremedia:///image/2656/data"
      ratio="landscape_ratio16x9"
      title="Globe"
    />
    <Overlay
      bottom="5%"
      color="blue"
      ctaShow={true}
      ctaText="Learn more"
      fixed={false}
      height="auto"
      key="overlay"
      text="Discover the new CoreMedia Headless Services."
      title="Content Management at a new Scale with CoreMedia CaaS"
      width={70}
    />
  </TeaserBox>
</OptionalLink>
`;

exports[`HeroTeaser Component should render HeroTeaser with Link, Picture and Overlay 2`] = `
<OptionalLink
  url="/caas.html"
>
  <TeaserBox>
    <WithTheme(Picture)
      alt="Globe"
      color="blue"
      key="picture"
      link="coremedia:///image/2656/data"
      ratio="landscape_ratio16x9"
      title="Globe"
    />
    <Overlay
      bottom="5%"
      color="blue"
      ctaShow={false}
      ctaText="Learn more"
      fixed={false}
      height="auto"
      key="overlay"
      text="Discover the new CoreMedia Headless Services."
      title="Content Management at a new Scale with CoreMedia CaaS"
      width={70}
    />
  </TeaserBox>
</OptionalLink>
`;

exports[`HeroTeaser Component should render HeroTeaser with Link, Picture and Overlay 3`] = `
<OptionalLink
  url="/caas.html"
>
  <TeaserBox>
    <WithTheme(Picture)
      alt="Globe"
      color="blue"
      key="picture"
      link="coremedia:///image/2656/data"
      ratio="landscape_ratio16x9"
      title="Globe"
    />
    <Overlay
      bottom="5%"
      color="blue"
      ctaShow={false}
      ctaText="Learn more"
      fixed={false}
      height="auto"
      key="overlay"
      title="Content Management at a new Scale with CoreMedia CaaS"
      width={70}
    />
  </TeaserBox>
</OptionalLink>
`;

exports[`HeroTeaser Component should render HeroTeaser with Link, Picture and Overlay 4`] = `
<OptionalLink
  url="/caas.html"
>
  <TeaserBox>
    <WithTheme(Picture)
      alt="Globe"
      color="blue"
      key="picture"
      link="coremedia:///image/2656/data"
      ratio="landscape_ratio16x9"
      title="Globe"
    />
    <Overlay
      bottom="5%"
      color="blue"
      ctaShow={false}
      ctaText="Learn more"
      fixed={false}
      height="auto"
      key="overlay"
      text="Discover the new CoreMedia Headless Services."
      width={70}
    />
  </TeaserBox>
</OptionalLink>
`;

exports[`HeroTeaser Component should render HeroTeaser with Link, Picture and Overlay 5`] = `
<OptionalLink
  url="/caas.html"
>
  <TeaserBox>
    <WithTheme(Picture)
      alt="Globe"
      color="blue"
      key="picture"
      link="coremedia:///image/2656/data"
      ratio="landscape_ratio16x9"
      title="Globe"
    />
    <Overlay
      bottom="5%"
      color="blue"
      ctaShow={true}
      ctaText="Learn more"
      fixed={false}
      height="auto"
      key="overlay"
      text="Discover the new CoreMedia Headless Services."
      width={70}
    />
  </TeaserBox>
</OptionalLink>
`;

exports[`HeroTeaser Component should render HeroTeaser with Link, Picture and Overlay 6`] = `
<OptionalLink
  url="/caas.html"
>
  <TeaserBox>
    <WithTheme(Picture)
      alt="Globe"
      color="blue"
      key="picture"
      link="coremedia:///image/2656/data"
      ratio="landscape_ratio16x9"
      title="Globe"
    />
    <Overlay
      bottom="5%"
      color="blue"
      ctaShow={true}
      ctaText="Learn more"
      fixed={false}
      height="auto"
      key="overlay"
      title="Content Management at a new Scale with CoreMedia CaaS"
      width={70}
    />
  </TeaserBox>
</OptionalLink>
`;

exports[`HeroTeaser Component should render HeroTeaser with Link, Picture and Overlay 7`] = `
<OptionalLink
  url="/caas.html"
>
  <TeaserBox>
    <WithTheme(Picture)
      alt="Globe"
      color="blue"
      key="picture"
      link="coremedia:///image/2656/data"
      ratio="landscape_ratio16x9"
      title="Globe"
    />
    <Overlay
      bottom="5%"
      color="blue"
      ctaShow={true}
      ctaText="Learn more"
      fixed={false}
      height="auto"
      key="overlay"
      width={70}
    />
  </TeaserBox>
</OptionalLink>
`;

exports[`HeroTeaser Component should render HeroTeaser with Overlay 1`] = `
<OptionalLink>
  <TeaserBox>
    <Overlay
      bottom="5%"
      color="blue"
      ctaShow={false}
      ctaText="Learn more"
      fixed={false}
      height="auto"
      key="overlay"
      width={70}
    />
  </TeaserBox>
</OptionalLink>
`;

exports[`HeroTeaser Component should render HeroTeaser with Picture (without params) 1`] = `
<OptionalLink>
  <TeaserBox>
    <WithTheme(Picture)
      alt="Globe"
      key="picture"
      link="coremedia:///image/2656/data"
      ratio="landscape_ratio16x9"
      title="Globe"
    />
  </TeaserBox>
</OptionalLink>
`;

exports[`HeroTeaser Component should render HeroTeaser with Picture 1`] = `
<OptionalLink>
  <TeaserBox>
    <WithTheme(Picture)
      alt="Globe"
      color="blue"
      key="picture"
      link="coremedia:///image/2656/data"
      ratio="landscape_ratio16x9"
      title="Globe"
    />
  </TeaserBox>
</OptionalLink>
`;

exports[`HeroTeaser Component should render HeroTeaser with Picture and Overlay 1`] = `
<OptionalLink>
  <TeaserBox>
    <WithTheme(Picture)
      alt="Globe"
      color="blue"
      key="picture"
      link="coremedia:///image/2656/data"
      ratio="landscape_ratio16x9"
      title="Globe"
    />
    <Overlay
      bottom="5%"
      color="blue"
      ctaShow={false}
      ctaText="Learn more"
      fixed={false}
      height="auto"
      key="overlay"
      width={70}
    />
  </TeaserBox>
</OptionalLink>
`;
